### ISSUE TITLE
SDK crash at app launch on iOS 12.0

### ISSUE THREAD

**Issue by oiledCode** [microsoft/appcenter-sdk-apple#1905]
### **Description**

The sdk crash at app launch on iOS 12.0

### **Repro Steps**

1. Integrate the SDK on an app
2. Initialize the SDK
3. Launch the app on a device running iOS 12.0
4. The SDK crash

### **Details**

The crash is generated by the call `-[MSDeviceTracker updateDevice]` line 145.
On this line the SDK try to access the `serviceSubscriberCellularProviders` property of a `CTTelephonyNetworkInfo` object, and although this property should be available on iOS 12.0 the system raise  an `unrecognized selector sent to instance xxxx` exception. 
This clearly smells like an iOS bug, but maybe can be mitigated by an `@available(iOS 12.1, *)` 
1. Which SDK version are you using?
    - e.g. 2.5.0
2. Which OS version did you experience the issue on?
    - e.g. iOS 12.0
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - e.g. iPhone phisical device
3. What language are you using?
    - [x] Objective C
    - [ ] Swift

**Comment by annakocheshkova** [comment]
Thanks for reporting that @oiledCode! I'll file a ticket for that.

**Comment by candostdagdeviren** [comment]
Hello! Is there any plan on fixing this soon? We're having the same crash on physical iPad with iOS 12. It is not reproducible on the simulator.

**Comment by ilyashumihin** [comment]
Hi, @candostdagdeviren!
This bug will be fixed in the next release - 2.5.4.

**Comment by Sai** [comment]
The #1914 seems not fix the issue because we have collected similar crashes on iOS 12.4.4.

The app might still crash if the SDK still calling serviceSubscriberCellularProviders above 12.1.

[Ref.](https://github.com/microsoft/appcenter-sdk-apple/blob/8887729f92de7a03ed3c03f59042f1dfe0438830/AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m#L147)
```
if (@available(iOS 12.1, *)) {
  NSDictionary<NSString *, CTCarrier *> *carriers = [telephonyNetworkInfo serviceSubscriberCellularProviders];
  carrier = [self firstCarrier:carriers];
}
```

**Comment by AnastasiaKubova** [comment]
Hi, @Sai ! Thanks for getting in touch with us! 
I try to reproduce this issue on `iPhone 6 Plus iOS 12.4.4`, but it works expected for me.
This fix will be released soon. We will notify you when the release is released.

**Comment by annakocheshkova** [comment]
Hi there! This issue has been fixed in the [3.0.0 release](https://github.com/microsoft/appcenter-sdk-apple/releases/tag/3.0.0). Please check it and reopen in case there are any issues.
